File: GameLibrary/src/cc/lib/reflector/Archivers.java
package cc.lib.reflector;

/**
 * Created by Chris Caron on 12/1/23.
 */
class Archivers {

    static Archiver byteArchiver = new AArchiver() {

        @Override
        public Object parse(String s) {
            return s.equals("null") ? null : Byte.parseByte(s);
        }

    };

    static Archiver integerArchiver = new AArchiver() {

        @Override
        public Object parse(String s) {
            return s.equals("null") ? null : Integer.parseInt(s);
        }

    };

    static Archiver longArchiver = new AArchiver() {

        @Override
        public Object parse(String s) {
            return s.equals("null") ? null : Long.parseLong(s);
        }

    };

    static Archiver floatArchiver = new AArchiver() {

        @Override
        public Object parse(String s) {
            return s.equals("null") ? null : Float.parseFloat(s);
        }

    };

    static Archiver doubleArchiver = new AArchiver() {

        @Override
        public Object parse(String s) {
            return s.equals("null") ? null : Double.parseDouble(s);
        }

    };

    static Archiver booleanArchiver = new AArchiver() {

        @Override
        public Object parse(String s) {
            return s.equals("null") ? null : Boolean.parseBoolean(s);
        }

    };

    static Archiver stringArchiver = new StringArchiver();

    static Archiver enumArchiver = new EnumArchiver();

    static Archiver dirtyArchiver = new DirtyArchiver();

    static Archiver collectionArchiver = new CollectionArchiver();

    static Archiver mapArchiver = new MapArchiver();

    static Archiver arrayArchiver = new ArrayArchiver();

    static Archiver archivableArchiver = new ArchivableArchiver();

}

File: GameLibrary/src/cc/lib/kreflector/Archivers.kt
package cc.lib.kreflector

import java.lang.Boolean
import kotlin.Any
import kotlin.String

/**
 * Created by Chris Caron on 12/1/23.
 */
val byteArchiver: Archiver = object : AArchiver() {
	override fun parse(s: String): Any {
		return (if (s == "null") null else s.toByte())!!
	}
}
val integerArchiver: Archiver = object : AArchiver() {
	override fun parse(s: String): Any {
		return (if (s == "null") null else s.toInt())!!
	}
}
val longArchiver: Archiver = object : AArchiver() {
	override fun parse(s: String): Any {
		return (if (s == "null") null else s.toLong())!!
	}
}
val floatArchiver: Archiver = object : AArchiver() {
	override fun parse(s: String): Any {
		return (if (s == "null") null else s.toFloat())!!
	}
}
val doubleArchiver: Archiver = object : AArchiver() {
	override fun parse(s: String): Any {
		return (if (s == "null") null else s.toDouble())!!
	}
}
val booleanArchiver: Archiver = object : AArchiver() {
	override fun parse(s: String): Any {
		return (if (s == "null") null else Boolean.parseBoolean(s))!!
	}
}
val stringArchiver: Archiver = StringArchiver()
val enumArchiver: Archiver = EnumArchiver()
val dirtyArchiver: Archiver = DirtyArchiver<Any?>()
val collectionArchiver: Archiver = CollectionArchiver()
val mapArchiver: Archiver = MapArchiver()
val arrayArchiver: Archiver = ArrayArchiver()
val archivableArchiver: Archiver = ArchivableArchiver()